spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -354,4 +355,13 @@ public TaskExecutionAwareTaskDefinition apply(TaskDefinition source) {
 			}
 		}
 	};
+
+	private void validateTaskName(String name) {
+		// Kubernetes does not support a pod name with length greater than 63 characters. It also
+		// adds 11 characters to the task name to create a pod, hence we need to ensure that task
+		// name added in SCDF has a length pf less than 52 characters.
+		if (name.length() > 52) {
```

Comment:
It turned out to be a simple length check, nice. I wonder if we need to handle this a bit more gracefully to preserve backward compatibility. There can be users on local/cf with a lengthier naming convention, and when they upgrade to the latest minor release (with this fix), it would break. @ilayaperumalg @jvalkeal @tzolov: Do you have any opinions?

Lastly, however we end up deciding to address it, let's please also apply it to stream names.

Reply:
We could add a validation step in the `KubernetesDeployerProoperties.deploy` method like what we do in `KuberetesScheduler.schedule`  method https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/blob/master/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesScheduler.java#L72-L74.

Reply:
The issue is that if we have more than one platform, we don't know which platform they will select until the task is launched.   i.e. it could be CF, Local or K8's.

Reply:
Yeah, that makes sense. So you are saying that we don't validate it when a user has added a task... But do it when it launches which is K8s specific?

Reply:
Correct.   Technically K8's already does this for us. Its just that the error message is cryptic for some users.   What I'm suggesting is that we make the error message more user friendly.

Reply:
So should I make that change in the other repo then?

Reply:
This topic brought up some good discussions in our last standup.     I'm chatting with some folks on this topic.   Again thanks for bringing this up and the contribution.